js/customize-snapshots.js

```diff
@@ -1,894 +1,971 @@
-	 */
-	component.init = function() {
-		window._wpCustomizeControlsL10n.save = component.data.i18n.publish;
-		window._wpCustomizeControlsL10n.saved = component.data.i18n.published;
-
-		// Set the initial client timestamp.
-		component.data.initialClientTimestamp = component.dateValueOf();
-
-		api.bind( 'ready', function() {
-			api.state.create( 'snapshot-exists', component.data.snapshotExists );
-			api.state.create( 'snapshot-saved', true );
-			api.state.create( 'snapshot-submitted', true );
-			api.bind( 'change', function() {
-				api.state( 'snapshot-saved' ).set( false );
-				api.state( 'snapshot-submitted' ).set( false );
-			} );
-			component.frontendPreviewUrl = new api.Value( api.previewer.previewUrl.get() );
-			component.frontendPreviewUrl.link( api.previewer.previewUrl );
-
-			component.extendPreviewerQuery();
-			component.addButtons();
-			component.editSnapshotUI();
-
-			$( '#snapshot-save' ).on( 'click', function( event ) {
-				var scheduleDate,
-					requestData = {
-						status: 'draft'
-					};
-				event.preventDefault();
+	var escKeyCode = 27;
 
-				if ( component.snapshotTitle && component.snapshotTitle.val() ) {
-					requestData.title = component.snapshotTitle.val();
-				}
-				if ( ! _.isEmpty( component.editContainer ) && component.isFutureDate() ) {
-					scheduleDate = component.getDateFromInputs();
-					requestData.status = 'future';
-					requestData.publish_date = component.formatDate( scheduleDate );
-					component.sendUpdateSnapshotRequest( requestData );
-				} else {
-					component.sendUpdateSnapshotRequest( requestData );
-				}
-			} );
-			$( '#snapshot-submit' ).on( 'click', function( event ) {
-				var requestData = {
-					status: 'pending'
-				};
-				event.preventDefault();
-				if ( component.snapshotTitle && component.snapshotTitle.val() ) {
-					requestData.title = component.snapshotTitle.val();
-				}
-				component.sendUpdateSnapshotRequest( requestData );
-			} );
+	api.Snapshots = api.Class.extend( {
+
+		data: {
+			action: '',
+			uuid: '',
+			editLink: '',
+			title: '',
+			publishDate: '',
+			postStatus: '',
+			currentUserCanPublish: true,
+			initialServerDate: '',
+			initialServerTimestamp: 0,
+			initialClientTimestamp: 0,
+			i18n: {},
+			dirty: false
+		},
+
+		uuidParam: 'customize_changeset_uuid',
+
+		initialize: function initialize( snapshotsConfig ) {
+			var snapshot = this, snapshotExists;
 
-			if ( api.state( 'snapshot-exists' ).get() ) {
-				api.state( 'saved' ).set( false );
-				component.resetSavedStateQuietly();
+			snapshot.schedule = {};
+
+			if ( _.isObject( snapshotsConfig ) ) {
+				_.extend( snapshot.data, snapshotsConfig );
 			}
 
-			api.trigger( 'snapshots-ready', component );
-		} );
+			// Set the initial client timestamp.
+			snapshot.data.initialClientTimestamp = snapshot.dateValueOf();
 
-		api.bind( 'save', function( request ) {
+			api.bind( 'ready', function() {
+				snapshotExists = '{}' !== api.previewer.query().customized;
```

Comment:
This isn't going to tell us whether the snapshot exists. It's just going to tell us whether there are dirty changes that aren't yet written into the changeset. If you want to check it the changeset exists, you should look at `api.state( 'changesetStatus' )` to see if it is not an empty string.